### PR TITLE
Slaughter Demon

### PR DESCRIPTION
**Pull request by ghost**
Had to recreate because I couldn't resolve the merge conflict.

New side event/antagonist. Simple animal that can use pools of blood as
portals to move around the station. Spooky scary.

-CtrlClick on an adjacent blood pool (or turf with blood on it) to
"enter" (start an infinite duration jaunt
-CtrlClick on a blood pool while jaunted to exit at that pool
-If you're pulling a mob while you enter a pool, the mob gets pulled in
with you, and you devour them, healing yourself
-Night vision

Adds an event (admin only) that spawns one.

Adds wizard item for summoning it. Has objective to kill the wizard (and everyone else).

Also fixes typo/filepath for the scrying orb in the new wizard datum thing.

If someone could help me with mind stuff so I can track how many people it eats that'd be great.

Webm

http://www.singulo.io/tg/src/1433549232359.webm

### CODE REVIEW

**Comment by KorPhaeron** [comment]
@RemieRichards @Miauw62

**Comment by Miauw62** [comment]
RR said this was ok to merge so hey

**Comment by KorPhaeron** [comment]
@Miauw62 

Travis is done

**Comment by Lumipharon** [comment]
Kor, would it not be possible for a wizard to load up on spells/apprentices/etc, summon a demon, then immediately mindswap it? (himself or an apprentice)

**Comment by KorPhaeron** [comment]
@Lumipharon

The demon spawns "phased out"/jaunted so not immediately no.

The wizard could probably hunt it down and swap later though (does it work on simple animals?) never considered that. Will probably have to add a check because a demon with fireball woulf be pretty ridiculous

**Comment by Lumipharon** [comment]
it doesn't work on simple animals, I assumed it was player controlled, so no issue then.

**Comment by KorPhaeron** [comment]
Its a player controlled simple animal so it'd work. I'll fix that when I do my first round of tweaks

**Comment by KorPhaeron** [comment]
Good catch, thanks

**Comment by Lumipharon** [comment]
Also on the topic of wizards, will the staff of change be able to demon people?

Summon rounds aren't really balanced, so it'd probably be worth a laugh.

Also, doesn't the slime reaction that creates simple mobs just generate random simple mobs, which would include demons? THAT would probably be bad.

**Comment by KorPhaeron** [comment]
I dont know why that doesnt work on a whitelist instead of blacklist this happens every time we add a new mob.

If we lived through xenobio being able to make ascendant shadowlings I think we should be okay for a couple days here as well. Need to fix it though

**Comment by Lumipharon** [comment]
the idea of a xenobiologist making an army of sentient demon friends is pretty neat.

**Comment by Anonus** [comment]
If they're making sentient demons, just hold those demons to their antag status and they'll stop doing it.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+
+////////////////////The Powers
+
+/mob/living/simple_animal/slaughter/proc/phaseout(var/obj/effect/decal/cleanable/B)
+	var/turf/mobloc = get_turf(src.loc)
+	var/turf/bloodloc = get_turf(B.loc)
+	if(Adjacent(bloodloc))
+		src.notransform = TRUE
+		spawn(0)
```
Comment (Aranclanos) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: why is there a spawn() here

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		src.loc = B.loc
+		src.phased = FALSE
+		src.client.eye = src
+		src.visible_message("<span class='warning'><B>The [src] rises out of the pool of blood!</B>")
+		playsound(get_turf(src), 'sound/magic/exit_blood.ogg', 100, 1, -1)
+		qdel(src.holder)
+
+/obj/effect/decal/cleanable/blood/CtrlClick(var/mob/user)
+	..()
+	if(istype(user, /mob/living/simple_animal/slaughter))
```
Comment (Aranclanos) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: remove these checks on these three procs

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	var/canmove = 1
+	density = 0
+	anchored = 1
+
+obj/effect/dummy/slaughter/relaymove(var/mob/user, direction)
+	if (!src.canmove || !direction) return
+	var/turf/newLoc = get_step(src,direction)
+	loc = newLoc
+	src.canmove = 0
+	spawn(1)
```
Comment (Aranclanos) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: another random spawn()

**Comment by Aranclanos** [comment]
the code of this is awful, how did it get merged?

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm (hunk 3 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: it's to prevent the mob from moving for 1 tick.
but sleep would work here too.

**Pull request merged**